### PR TITLE
[routing-manager] remove vicarious router discovery feature

### PR DESCRIPTION
**Pull request by jwhui**
The vicarious router discovery feature depends on the ability to
receive Router Advertisement messages sent in response to Router
Solicitation messages from other hosts. However, this does not work
since Router Advertisement messages are often sent unicast in
response.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM.

File: tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py [hunk 1]
```diff
@@ -1,216 +0,0 @@
...
-ROUTER1 = 2
-BR2 = 3
-ROUTER2 = 4
-HOST = 5
-
-CHANNEL1 = 18
-CHANNEL2 = 19
-
-
-class MultiThreadNetworks(thread_cert.TestCase):
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py:59]: If we leave the test alone, it can still pass.

The test seems unrelated to `OPENTHREAD_CONFIG_BORDER_ROUTING_VICARIOUS_RS_ENABLE` from what it does. 

Maybe we can simply keep and rename the test? @jwhui @wgtdkp

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

File: tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py (hunk 1 above)
Comment (jwhui) [reply tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py]: I renamed this test to `test_on_link_prefix.py`. Thoughts?

Comment (simonlingoogle) [reply tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py]: SGTM.

**Review by wgtdkp — APPROVED** [review]
LGTM 👍
